targets/psa_switch/tests/test_internet_checksum.cpp

```diff
@@ -0,0 +1,231 @@
+                            get_primitive("_InternetChecksum_subtract");
+    ASSERT_NE(nullptr, primitive_sub);
+    actionFn_sub.push_back_primitive(primitive_sub.get());
+    actionFn_sub.parameter_push_back_extern_instance(instance.get());
+    actionFn_sub.parameter_start_field_list();
+    actionFn_sub.parameter_push_back_field(ipv4Header, 10);
+    actionFn_sub.parameter_push_back_field(ipv4Header, 11);
+    actionFn_sub.parameter_end_field_list();
+
+    // ACTION GET_STATE
+    ActionFn actionFn_get_state("_InternetChecksum_get_state", 0, 0);
+    ActionFnEntry actionFnEntry_get_state(&actionFn_get_state);
+    auto primitive_get_state =  ActionOpcodesMap::get_instance()->
+                            get_primitive("_InternetChecksum_get_state");
+    ASSERT_NE(nullptr, primitive_get_state);
+    actionFn_get_state.push_back_primitive(primitive_get_state.get());
+    actionFn_get_state.parameter_push_back_extern_instance(instance.get());
+    actionFn_get_state.parameter_push_back_field(metaHeader, 0);
+
+    // ACTION SET_STATE
+    ActionFn actionFn_set_state("_InternetChecksum_set_state", 0, 0);
+    ActionFnEntry actionFnEntry_set_state(&actionFn_set_state);
+    auto primitive_set_state =  ActionOpcodesMap::get_instance()->
+                            get_primitive("_InternetChecksum_set_state");
+    ASSERT_NE(nullptr, primitive_set_state);
+    actionFn_set_state.push_back_primitive(primitive_set_state.get());
+    actionFn_set_state.parameter_push_back_extern_instance(instance.get());
+    actionFn_set_state.parameter_push_back_const(Data("0x01"));
+
+    cksum = 0xb6ab;
+    sum = 0x4954;
+
+    actionFnEntry_add(packet.get());
+    actionFnEntry_get(packet.get());
+    actionFnEntry_get_state(packet.get());
+    ASSERT_EQ(sum, phv->get_field("meta.tmp").get<uint16_t>());
+    ASSERT_EQ(cksum, phv->get_field("ipv4.checksum").get<uint16_t>());
+
+    cksum = 0xb6ae;
+    sum =  0x4951;
+
+    actionFnEntry_sub(packet.get());
+    actionFnEntry_get(packet.get());
+    actionFnEntry_get_state(packet.get());
+    ASSERT_EQ(sum, phv->get_field("meta.tmp").get<uint16_t>());
+    ASSERT_EQ(cksum, phv->get_field("ipv4.checksum").get<uint16_t>());
+
+    sum = 0x01;
+
+    actionFnEntry_set_state(packet.get());
+    actionFnEntry_get_state(packet.get());
+    ASSERT_EQ(sum, phv->get_field("meta.tmp").get<uint16_t>());
+}
+
+BM_REGISTER_EXTERN_W_NAME(InternetChecksum, PSA_InternetChecksum);
+BM_REGISTER_EXTERN_W_NAME_METHOD(InternetChecksum, PSA_InternetChecksum, add, const std::vector<Field>);
+BM_REGISTER_EXTERN_W_NAME_METHOD(InternetChecksum, PSA_InternetChecksum, subtract, const std::vector<Field>);
+BM_REGISTER_EXTERN_W_NAME_METHOD(InternetChecksum, PSA_InternetChecksum, get, Field &);
+BM_REGISTER_EXTERN_W_NAME_METHOD(InternetChecksum, PSA_InternetChecksum, get_state, Field &);
+BM_REGISTER_EXTERN_W_NAME_METHOD(InternetChecksum, PSA_InternetChecksum, set_state,const Data &);
```

Comment:
calling `import_internet_checksum()` somewhere during test set-up should work, and you wouldn't need to duplicate those IIRC